libraries/FreeRTOS/src/FreeRTOSConfig.h

```diff
@@ -1,104 +1,231 @@
-#define configGENERATE_RUN_TIME_STATS	1
 #define portCONFIGURE_TIMER_FOR_RUN_TIME_STATS() //vMainConfigureTimerForRunTimeStats()
 #define portGET_RUN_TIME_COUNTER_VALUE() ulMainGetRunTimeCounterValue()
+#endif
 
 /* Co-routine definitions. */
+#ifndef configUSE_CO_ROUTINES
 #define configUSE_CO_ROUTINES           0
+#endif
+#ifndef configMAX_CO_ROUTINE_PRIORITIES
 #define configMAX_CO_ROUTINE_PRIORITIES ( 2 )
+#endif
 
 /* Software timer definitions. */
+#ifndef configUSE_TIMERS
 #define configUSE_TIMERS             1
+#endif
+#ifndef configTIMER_TASK_PRIORITY
 #define configTIMER_TASK_PRIORITY    ( 2 )
+#endif
+#ifndef configTIMER_QUEUE_LENGTH
 #define configTIMER_QUEUE_LENGTH     5
+#endif
+#ifndef configTIMER_TASK_STACK_DEPTH
 #define configTIMER_TASK_STACK_DEPTH ( 1024 )
+#endif
 
 /*  Set the following definitions to 1 to include the API function, or zero
     to exclude the API function. */
+#ifndef INCLUDE_eTaskGetState
 #define INCLUDE_eTaskGetState                   1
+#endif
+#ifndef INCLUDE_uxTaskGetStackHighWaterMark
 #define INCLUDE_uxTaskGetStackHighWaterMark 	1
+#endif
+#ifndef INCLUDE_uxTaskPriorityGet
 #define INCLUDE_uxTaskPriorityGet               1
+#endif
+#ifndef INCLUDE_vTaskCleanUpResources
 #define INCLUDE_vTaskCleanUpResources           1
+#endif
+#ifndef INCLUDE_vTaskDelay
 #define INCLUDE_vTaskDelay                      1
+#endif
+#ifndef INCLUDE_vTaskDelayUntil
 #define INCLUDE_vTaskDelayUntil                 1
+#endif
+#ifndef INCLUDE_vTaskDelete
 #define INCLUDE_vTaskDelete                     1
+#endif
+#ifndef INCLUDE_vTaskPrioritySet
 #define INCLUDE_vTaskPrioritySet                1
+#endif
+#ifndef INCLUDE_vTaskSuspend
 #define INCLUDE_vTaskSuspend                    1
+#endif
+#ifndef INCLUDE_xQueueGetMutexHolder
 #define INCLUDE_xQueueGetMutexHolder            1
+#endif
+#ifndef INCLUDE_xTaskAbortDelay
 #define INCLUDE_xTaskAbortDelay                 1
+#endif
+#ifndef INCLUDE_xTaskGetCurrentTaskHandle
 #define INCLUDE_xTaskGetCurrentTaskHandle       1
+#endif
+#ifndef INCLUDE_xTaskGetHandle
 #define INCLUDE_xTaskGetHandle                  1
+#endif
+#ifndef INCLUDE_xTaskGetIdleTaskHandle
 #define INCLUDE_xTaskGetIdleTaskHandle          1
+#endif
+#ifndef INCLUDE_xTaskGetSchedulerState
 #define INCLUDE_xTaskGetSchedulerState          1
+#endif
+#ifndef INCLUDE_xTaskResumeFromISR
 #define INCLUDE_xTaskResumeFromISR              1
+#endif
+#ifndef INCLUDE_xTimerPendFunctionCall
 #define INCLUDE_xTimerPendFunctionCall          1
+#endif
 
+#ifndef configUSE_STATS_FORMATTING_FUNCTIONS
 #define configUSE_STATS_FORMATTING_FUNCTIONS	1
+#endif
 
+#ifndef configUSE_MUTEXES
 #define configUSE_MUTEXES              1
+#endif
+#ifndef configUSE_MALLOC_FAILED_HOOK
 #define configUSE_MALLOC_FAILED_HOOK   1
+#endif
+#ifndef configCHECK_FOR_STACK_OVERFLOW
 #define configCHECK_FOR_STACK_OVERFLOW 2
+#endif
 
 /* Cortex-M specific definitions. */
+#ifndef configPRIO_BITS
 #undef __NVIC_PRIO_BITS
 #ifdef __NVIC_PRIO_BITS
```

Comment:
Granted I cut-n-paste this file (or more likely it came already cut-n-pasted from upstream) but the sequence

````
#undef X
#ifdef X
````
is just wrong on so many levels. :laughing: